pkg/apis/stork/v1alpha1/types.go

```diff
@@ -212,3 +213,81 @@ type MigrationList struct {
 
 	Items []Migration `json:"items"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// GroupVolumeSnapshot represents a group snapshot
+type GroupVolumeSnapshot struct {
+	meta.TypeMeta   `json:",inline"`
+	meta.ObjectMeta `json:"metadata,omitempty"`
+	Spec            GroupVolumeSnapshotSpec   `json:"spec"`
+	Status          GroupVolumeSnapshotStatus `json:"status"`
+}
+
+// GroupVolumeSnapshotSpec represents the spec for a group snapshot
+type GroupVolumeSnapshotSpec struct {
+	// PreSnapshotRule is the name of rule applied before taking the snapshot. The rule needs to be
+	// in the same namespace as the group volumesnapshot
+	PreSnapshotRule string `json:"preSnapshotRule"`
+	// PreSnapshotRule is the name of rule applied after taking the snapshot. The rule needs to be
+	// in the same namespace as the group volumesnapshot
+	PostSnapshotRule string `json:"postSnapshotRule"`
+	// PVCSelector selects the PVCs that are part of the group snapshot
+	PVCSelector PVCSelectorSpec `json:"pvcSelector"`
+	// Options are pass-through parameters that are passed to the driver handling the group snapshot
+	Options map[string]string `json:"options"`
+}
+
+// PVCSelectorSpec is the spec to select the PVCs for group snapshot
+type PVCSelectorSpec struct {
+	meta.LabelSelector
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// GroupVolumeSnapshotList is a list of group volume snapshots
+type GroupVolumeSnapshotList struct {
+	meta.TypeMeta `json:",inline"`
+	meta.ListMeta `json:"metadata,omitempty"`
+
+	Items []GroupVolumeSnapshot `json:"items"`
+}
+
+// GroupVolumeSnapshotStatus is status for the group snapshot
+type GroupVolumeSnapshotStatus struct {
+	Stage           GroupVolumeSnapshotStageType  `json:"stage"`
+	Status          GroupVolumeSnapshotStatusType `json:"status"`
+	VolumeSnapshots []*crdv1.VolumeSnapshot       `json:"volumeSnapshots"`
```

Comment:
I don't think this needs the entire CRD here, just the name of the snapshots and their status should suffice.